packages/models/src/eservice/protobufConverterToV2.ts

```diff
@@ -0,0 +1,115 @@
+import { P, match } from "ts-pattern";
+import {
+  AgreementApprovalPolicyV2,
+  EServiceAttributeV2,
+  EServiceDescriptorStateV2,
+  EServiceDescriptorV2,
+  EServiceDocumentV2,
+  EServiceModeV2,
+  EServiceRiskAnalysisV2,
+  EServiceTechnologyV2,
+  EServiceV2,
+} from "../gen/v2/eservice/eservice.js";
+import { RiskAnalysis } from "../risk-analysis/riskAnalysis.js";
+import {
+  AgreementApprovalPolicy,
+  Descriptor,
+  DescriptorState,
+  Document,
+  EService,
+  EServiceAttribute,
+  EServiceMode,
+  Technology,
+} from "./eservice.js";
+
+export const toAgreementApprovalPolicyV2 = (
+  input: AgreementApprovalPolicy | undefined
+): AgreementApprovalPolicyV2 =>
+  match(input)
+    .with(P.nullish, () => AgreementApprovalPolicyV2.AUTOMATIC)
+    .with("Manual", () => AgreementApprovalPolicyV2.MANUAL)
+    .with("Automatic", () => AgreementApprovalPolicyV2.AUTOMATIC)
+    .exhaustive();
+
+export const toEServiceDescriptorStateV2 = (
+  input: DescriptorState
+): EServiceDescriptorStateV2 =>
+  match(input)
+    .with("Draft", () => EServiceDescriptorStateV2.DRAFT)
+    .with("Suspended", () => EServiceDescriptorStateV2.SUSPENDED)
+    .with("Archived", () => EServiceDescriptorStateV2.ARCHIVED)
+    .with("Published", () => EServiceDescriptorStateV2.PUBLISHED)
+    .with("Deprecated", () => EServiceDescriptorStateV2.DEPRECATED)
+    .exhaustive();
+
+export const toEServiceTechnologyV2 = (
+  input: Technology
+): EServiceTechnologyV2 =>
+  match(input)
+    .with("Rest", () => EServiceTechnologyV2.REST)
+    .with("Soap", () => EServiceTechnologyV2.SOAP)
+    .exhaustive();
+
+export const toEServiceModeV2 = (input: EServiceMode): EServiceModeV2 =>
+  match(input)
+    .with("Deliver", () => EServiceModeV2.DELIVER)
+    .with("Receive", () => EServiceModeV2.RECEIVE)
+    .exhaustive();
```

Comment:
@galales should we replace all the hardcoded strings with their dynamic values?
For example:
`.with(descriptorState.draft, () => EServiceDescriptorStateV2.DRAFT)`
instead of
`.with("Draft", () => EServiceDescriptorStateV2.DRAFT)`

Reply:
sounds good 👍

Reply:
Updated